langchain4j-mongodb-atlas/src/test/java/dev/langchain4j/store/embedding/mongodb/MongoDbEmbeddingStoreMiscIT.java

```diff
@@ -0,0 +1,122 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.model.Filters;
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingSearchRequest;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.mongodb.MongoDBAtlasLocalContainer;
+
+import java.time.Duration;
+import java.util.List;
+
+import static dev.langchain4j.store.embedding.TestUtils.awaitUntilAsserted;
+import static dev.langchain4j.store.embedding.mongodb.MongoDbTestFixture.*;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.data.Percentage.withPercentage;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class MongoDbEmbeddingStoreMiscIT {
+
+    public static class ContainerIT extends MongoDbEmbeddingStoreMiscIT {
+        static MongoDBAtlasLocalContainer mongodb = new MongoDBAtlasLocalContainer("mongodb/mongodb-atlas-local:7.0.9");
+
+        @BeforeAll
+        static void start() {
+            MongoDbTestFixture.assertDoContainerTests();
+            mongodb.start();
+        }
+
+        @AfterAll
+        static void stop() {
+            mongodb.stop();
+        }
+
+        @Override
+        MongoClient createClient() {
+            return createClientFromContainer(mongodb);
+        }
```

Comment:
Move to TestMongoDb

Reply:
Could you elaborate on how this (and the other suggestions related to moving) would work? Note that `createClient` is an override, which differs between tests. The static nested classes `ContainerIT` extend different classes, and I do not see how the duplication can be removed, beyond what is already done in this PR.

Reply:
@katcharov , for integration testing it is usually make sense to start containers once. With JUnit 5 it can be achieved by creating a singleton, representing external resource, like database container or wiremock server, so JVM guarantees that it it initialized once on the first use. Helper methods, like retrieving DB client should be also moved to this singletnon, because it is aware about connection details.

Reply:
I believe the container is started only once, in the fixture, and is then used across all tests.